common/logging/include/log-defines.h

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright IBM Corp. All Rights Reserved.
+ * Copyright 2020 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef LOG_DEFINES
+#define LOG_DEFINES
+
+#ifndef TAG
+#define TAG ""
+#endif
+
+#define LOC_FMT " (%s:%d) "
+
+#define NRM "\x1B[0m"
+#define CYN "\x1B[36m"
+#define YEL "\x1B[33m"
+#define RED "\x1B[31m"
+
+#ifndef DO_DEBUG
+#define DO_DEBUG false
+#endif
+
+#ifndef DO_INFO
+#define DO_INFO true
+#endif
+
+#ifndef DO_WARNING
+#define DO_WARNING true
+#endif
+
+#ifndef DO_ERROR
+#define DO_ERROR true
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+int loggingf(const char* fmt, ...);
+#ifdef __cplusplus
+}
+#endif
+
+#if DO_DEBUG == true
+#define LOG_DEBUG(fmt, ...) \
+    loggingf(CYN "DEBUG " LOC_FMT TAG YEL fmt NRM "\n", __FILE__, __LINE__, ##__VA_ARGS__)
```

Comment:
Passing the logging level explicitly rather than implicitly inside the string would in principle allow you do better output filtering. That said, given that the flogger support in the go stub/peer is essentially dropped i guess it really doesn't matter. And need arises we obviously can add that at that time ...

Reply:
I agree.